client/ayon_core/hosts/traypublisher/plugins/create/create_csv_ingest.py

```diff
@@ -0,0 +1,722 @@
+import os
+import re
+import csv
+import clique
+from copy import deepcopy, copy
+
+from ayon_core.client import get_asset_by_name
+from ayon_core.pipeline.create import get_product_name
+from ayon_core.pipeline import CreatedInstance
+from ayon_core.lib import FileDef, BoolDef
+from ayon_core.lib.transcoding import (
+    VIDEO_EXTENSIONS, IMAGE_EXTENSIONS
+)
+
+from ayon_core.hosts.traypublisher.api.plugin import (
+    TrayPublishCreator
+)
+
+
+class IngestCSV(TrayPublishCreator):
+    """CSV ingest creator class"""
+
+    icon = "fa.file"
+
+    label = "CSV Ingest"
+    product_type = "csv_ingest_file"
+    identifier = "io.ayon_core.creators.traypublisher.csv_ingest"
+
+    default_variants = ["Main"]
+
+    description = "Ingest products' data from CSV file"
+    detailed_description = """
+Ingest products' data from CSV file following column and representation
+configuration in project settings.
+"""
+
+    # Position in the list of creators.
+    order = 10
+
+    # settings for this creator
+    columns_config = {}
+    representations_config = {}
+
+    def create(self, subset_name, instance_data, pre_create_data):
+        """Create an product from each row found in the CSV.
+
+        Args:
+            subset_name (str): The subset name.
+            instance_data (dict): The instance data.
+            pre_create_data (dict):
+        """
+
+        csv_filepath_data = pre_create_data.get("csv_filepath_data", {})
+
+        folder = csv_filepath_data.get("directory", "")
+        if not os.path.exists(folder):
+            raise FileNotFoundError(
```

Comment:
This would show traceback to user, rather us `CreatorError` (`from ayon_core.pipeline.create import CreatorError`).